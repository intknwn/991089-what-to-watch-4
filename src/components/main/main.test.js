import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Main from './main.jsx';

const promoMovie = {
  name: `Die Hard`,
  previewImg: `img/die-hard.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Action`,
  year: 1988,
  rating: 8.2,
  score: 796619,
  description: `An NYPD officer tries to save his wife and several others taken hostage by German terrorists during a Christmas party at the Nakatomi Plaza in Los Angeles.`,
  director: `John McTiernan`,
  cast: [`Bruce Willis`, `Alan Rickman`, `Bonnie Bedelia`],
  runtime: 112,
};

const movies = [{
  id: 1,
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  previewImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Fantasy`,
  year: 2018,
  rating: 6.6,
  score: 240,
  description: `In an effort to thwart Grindelwald's plans of raising pure-blood wizards to rule over all non-magical beings, Albus Dumbledore enlists his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world.`,
  director: `David Yates`,
  cast: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
},
{
  id: 2,
  name: `Bohemian Rhapsody`,
  previewImg: `img/bohemian-rhapsody.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Biography`,
  year: 2018,
  rating: 8.0,
  score: 450,
  description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
  director: `Bryan Singer`,
  cast: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
},
{
  id: 3,
  name: `We need to talk about Kevin`,
  previewImg: `img/we-need-to-talk-about-kevin.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Thriller`,
  year: 2011,
  rating: 7.5,
  score: 138,
  description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
  director: `Lynne Ramsay`,
  cast: [`Tilda Swinton`, `John C. Reilly`, `Ezra Miller`],
},
{
  id: 4,
  name: `What We Do in the Shadows`,
  previewImg: `img/what-we-do-in-the-shadows.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Comedy`,
  year: 2019,
  rating: 8.5,
  score: 36894,
  description: `A look into the daily (or rather, nightly) lives of three vampires, who've lived together for over 100 years, on Staten Island.`,
  director: `Jemaine Clement`,
  cast: [`Kayvan Novak`, `Matt Berry`, `Natasia Demetriou`],
}];

const genres = [`All genres`, `Comedy`, `Adventure`, `Crime`, `Drama`, `Action`, `Thriller`, `Fantasy`];

const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarSrc: `/img/1.png`,
};

it(`Main component should render main screen`, () => {

  const tree = renderer
      .create(
          <BrowserRouter>
            <Main
              genres={genres}
              isAuthorized={true}
              movies={movies}
              promoMovie={promoMovie}
              setFavoriteStatus={() => {}}
              user={user}
            />
          </BrowserRouter>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
