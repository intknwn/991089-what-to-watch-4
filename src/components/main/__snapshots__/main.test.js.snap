// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render main screen 1`] = `
<div>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Die Hard"
        src="https://via.placeholder.com/1300x552"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href=""
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <img
              alt="Oliver.conner@gmail.com"
              height={63}
              src="https://4.react.pages.academy/img/1.png"
              width={63}
            />
          </a>
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Die Hard poster"
            height={327}
            src="https://via.placeholder.com/273x410"
            width={218}
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Die Hard
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              1988
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={19}
                viewBox="0 0 19 19"
                width={19}
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height={20}
                viewBox="0 0 19 20"
                width={19}
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Adventure
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thriller
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Fantasy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height={175}
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movies/1"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Bohemian Rhapsody"
              height={175}
              src="img/bohemian-rhapsody.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movies/2"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="We need to talk about Kevin"
              height={175}
              src="img/we-need-to-talk-about-kevin.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movies/3"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="What We Do in the Shadows"
              height={175}
              src="img/what-we-do-in-the-shadows.jpg"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/movies/4"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
